lua/lspconfig/server_configurations/solc.lua

```diff
@@ -0,0 +1,27 @@
+local util = require 'lspconfig.util'
+
+local bin_name = 'solc-static-linux'
+if vim.fn.has 'mac' then
+  bin_name = 'solc-macos'
+end
+if vim.fn.has 'win32' == 1 then
+  bin_name = 'solc-windows'
+end
```

Comment:
This assumes installation via prebuilt binaries distributed via GitHub releases (https://github.com/ethereum/solidity/releases).

Reply:
In nvim-lsp-installer these get normalized to `solc(.exe)`. I have not properly read through https://docs.soliditylang.org/en/latest/installing-solidity.html#building-from-source to know if there's a canonical name for it

Reply:
The canonical name would be `solc`:
https://docs.soliditylang.org/en/latest/using-the-compiler.html#basic-usage